shenyu-plugin/shenyu-plugin-uri/src/main/java/org/apache/shenyu/plugin/uri/URIPlugin.java

```diff
@@ -42,27 +42,18 @@
         if (StringUtils.isBlank(path)) {
             return chain.execute(exchange);
         }
-        String rewriteURI = (String) exchange.getAttributes().get(Constants.REWRITE_URI);
+        String rewriteUri = (String) exchange.getAttributes().get(Constants.REWRITE_URI);
         URI uri = exchange.getRequest().getURI();
-        if (StringUtils.isNoneBlank(rewriteURI)) {
-            path = path + rewriteURI;
+        if (StringUtils.isNoneBlank(rewriteUri)) {
+            path = path + rewriteUri;
         } else {
             String realUrl = shenyuContext.getRealUrl();
             if (StringUtils.isNoneBlank(realUrl)) {
                 path = path + realUrl;
             }
         }
-        URI realURI;
-        if (StringUtils.isNotEmpty(uri.getRawQuery()) && uri.getRawQuery().contains("%")) {
-            path = String.join("?", path, uri.getRawQuery());
-            realURI = UriComponentsBuilder.fromHttpUrl(path).build(true).toUri();
-        } else {
-            if (StringUtils.isNotEmpty(uri.getQuery())) {
-                path = String.join("?", path, uri.getQuery());
-            }
-            realURI = UriComponentsBuilder.fromHttpUrl(path).build(false).toUri();
-        }
-        exchange.getAttributes().put(Constants.HTTP_URI, realURI);
+        URI realUri = URI.create(path + "?" + RequestQueryCodecUtil.getCodecQuery(exchange));
```

Comment:
why modify this， only used  UriUtils.encode(item, StandardCharsets.UTF_8))？

Reply:
some query like `status=[abc%2Cedf]` will catch the `Invalid character '[' for QUERY_PARAM in status=[abc%2Cedf]`

Reply:
I will remove the `encode` transfer of the original data to upstream.